packages/ComponentLibrary/src/hooks/useItemType.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react';
+import { Menu } from '@workspaceui/etendohookbinder/src/api/types';
+
+interface UseItemActionsProps {
+  onWindowClick?: (windowId: string) => void;
+  onReportClick?: (reportId: string, recordId: string) => void;
+  onProcessClick?: (processId: string) => void;
+}
+
+export const useItemActions = ({ onWindowClick, onReportClick, onProcessClick }: UseItemActionsProps) => {
+  const getReportRecordId = useCallback((item: Menu) => {
+    if (item.recordId) {
+      return item.recordId;
+    }
+
+    if (item.windowId) {
+      return item.windowId;
+    }
+
+    if (item.tableId) {
+      return item.tableId;
+    }
+
+    return item.id;
+  }, []);
```

Comment:
please move this outside of the hook (it doesn't need to be defined as a callback inside of the hook, it's a regular helper)